miner/worker.go

```diff
@@ -259,10 +259,24 @@ func (w *worker) update() {
 			span.End()
 
 		// Handle NewTxsEvent
-		case ev, ok := <-w.txsCh:
+		case first, ok := <-w.txsCh:
 			if !ok {
 				return
 			}
+			evs := []core.NewTxsEvent{first}
+			// Check for more ready events.
+			for len(evs) < 1000 {
+				select {
+				case ev, ok := <-w.txsCh:
+					if !ok {
+						return
+					}
+					evs = append(evs, ev)
+				default:
+					break
+				}
+			}
```

Comment:
The `break` inside the `default:` will apply to the `select` and not the `for` loop. You'll need a label on the `for` loop to break out of that.

https://golang.org/ref/spec#Break_statements